lambda_functions/delivery_status_processor_lambda/delivery_status_processor_lambda.py

```diff
@@ -34,12 +36,58 @@
     logger.warning("Invalid log level specified, defaulting to INFO")
 
 
+def get_twilio_token():
+    try:
+        ssm_client = boto3.client("ssm", "us-gov-west-1")
+        auth_ssm_key = os.getenv("TWILIO_AUTH_TOKEN_SSM_NAME", "")
+        if not auth_ssm_key:
+            logger.error("TWILIO_AUTH_TOKEN_SSM_NAME not set")
+
+        response = ssm_client.get_parameter(
+            Name=auth_ssm_key,
+            WithDecryption=True
+        )
+        return response.get("Parameter").get("Value")
+    except Exception as e:
+        logger.error('Failed to retrieve Twilio Auth')
+        return None
+
+
+auth_token = get_twilio_token()
```

Comment:
I'd ask Kyle about this, but I feel like it doesn't follow the standard set above for setting up the environment. It seems like the following is more like what we have already done:

right after the other env vars:
`TWILIO_AUTH_TOKEN_SSM_NAME = os.getenv("DELIVERY_STATUS_RESULT_TASK_QUEUE")`

right after the other env var checks for None
```
if TWILIO_AUTH_TOKEN_SSM_NAME is None:
    sys.exit("A required environment variable is not set. Please set TWILIO_AUTH_TOKEN_SSM_NAME")
```
Because if this doesn't exist, we don't even want the lambda to do anything.

Then, right after where `sqs_client` is set:

```
ssm_client = boto3.client("ssm", "us-gov-west-1")
try:
    response = ssm_client.get_parameter(
        Name=auth_ssm_key,
        WithDecryption=True
        )
    auth_token = response.get("Parameter").get("Value")
except Exception as e:
    sys.exit("Unable to obtain the auth token from AWS. Please check the secret.")
```
Because again if we don't have an auth token, then the lambda should just stop.

Then proceed with setting up logging and the code. Again, I'd check with Kyle here.

Reply:
Cris (and Kyle), 

Contextually, the usage of SSM and SQS are rather different in this context. SSM is called once at initiation and available to all subsequent calls to the `delivery_status_processor_lambda_handler`. Call once, use often. Kyle called this out, hence the change. 

Conversely, SQS has to be called each time we put something on the queue because each call is unique. 

Moreover, we only use the SSM in the case of Twilio requests whereas this lambda handles Pinpoint and SQS calls as well. All 3 use cases use SQS, only the Twilio case needs SMS. The intention is that if we break the Twilio functionality (which we should not), the other functionalities in this Lambda would carry on happily while we fix the code. If this were to happen (better not) then we would have less messages to remediate if the lambda as a whole carries on if missing a variable only needed for 1/3 of it's use cases.

Reply:
Good points! Thank you.